lms/djangoapps/learner_dashboard/urls.py

```diff
@@ -6,5 +6,6 @@
 from . import views
 
 urlpatterns = [
+    url(r'^programs/(?P<programId>[0-9]+)/$', views.view_program_details, name='program_details_view'),
```

Comment:
https://en.wikipedia.org/wiki/Universally_unique_identifier.
I believe in URL, we use the canonical form